cedar-policy/src/api/err.rs

```diff
@@ -112,6 +113,210 @@ pub enum ReAuthorizeError {
     },
 }
 
+/// The kind of hierarchy that has transitive closure errors
+#[derive(Debug, Clone)]
+pub enum HierarchyKind {
+    /// An action hierarchy
+    Action,
+    /// An entity type hierarchy
+    EntityType,
+}
+
+impl std::fmt::Display for HierarchyKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Action => write!(f, "action"),
+            Self::EntityType => write!(f, "entity type"),
+        }
+    }
+}
+
+/// Transitive closure computation or enforcement error
+#[derive(Debug, Diagnostic, Error, Clone)]
+#[error("transitive closure computation/enforcement error on {kind} hierarchy: {message}")]
+pub struct TransitiveClosureError {
+    kind: HierarchyKind,
+    message: SmolStr,
+}
+
+impl TransitiveClosureError {
+    /// Getter of hierarchy kind
+    pub fn get_kind(&self) -> HierarchyKind {
+        self.kind.clone()
+    }
+
+    /// Getter of error message
+    pub fn get_message(&self) -> SmolStr {
+        self.message.clone()
```

Comment:
I think I'd rather be conservative with what internal details are `pub` for error types. 

`message` specifically feels like it should be definitly hidden implementation detail.